### ISSUE TITLE
Infinite recursion runtime in plumbing

### ISSUE THREAD

**Issue by kriskog** [tgstation/tgstation#47392]
## Round ID:
122174

```
[2019-10-25 13:33:31.424] runtime error: Maximum recursion level reached (perhaps there is an infinite loop)
 - To avoid this safety check, set world.loop_checks=0.
 - proc name: GetComponent (/datum/proc/GetComponent)
 -   usr: Brick Carr (/mob/living/carbon/human)
 -   src:  (/atom/movable/lighting_object)
 -   usr.loc: the carpet (82,127,2) (/turf/open/floor/carpet)
 -   src.loc: the floor (56,101,2) (/turf/open/floor/plasteel/white)
 -   call stack:
 -  (/atom/movable/lighting_object): GetComponent(/datum/component/plumbing (/datum/component/plumbing))
 - the fluid duct (/obj/machinery/duct): connect network( (/atom/movable/lighting_object), 1, null)
 - the fluid duct (/obj/machinery/duct): attempt connect()
 - the fluid duct (/obj/machinery/duct): connect duct(the fluid duct (/obj/machinery/duct), 8, null)
 - the fluid duct (/obj/machinery/duct): connect network(the fluid duct (/obj/machinery/duct), 8, null)
 - the fluid duct (/obj/machinery/duct): attempt connect()
 - the fluid duct (/obj/machinery/duct): connect duct(the fluid duct (/obj/machinery/duct), 8, null)
 - the fluid duct (/obj/machinery/duct): connect network(the fluid duct (/obj/machinery/duct), 8, null)
 - the fluid duct (/obj/machinery/duct): attempt connect()
 - the fluid duct (/obj/machinery/duct): connect duct(the fluid duct (/obj/machinery/duct), 8, null)
 - ...
 - the fluid duct (/obj/machinery/duct): connect duct(the fluid duct (/obj/machinery/duct), 4, null)
 - the fluid duct (/obj/machinery/duct): attempt connect()
 - the fluid duct (/obj/machinery/duct): connect network(the fluid duct (/obj/machinery/duct), 4, null)
 - the fluid duct (/obj/machinery/duct): connect duct(the fluid duct (/obj/machinery/duct), 2, null)
 - the fluid duct (/obj/machinery/duct): attempt connect()
 - the fluid duct (/obj/machinery/duct): disconnect duct()
 - the fluid duct (/obj/machinery/duct): tool act(Brick Carr (/mob/living/carbon/human), the wrench (/obj/item/wrench), "wrench")
 - Brick Carr (/mob/living/carbon/human): ClickOn(the fluid duct (/obj/machinery/duct), "icon-x=21;icon-y=16;left=1;scr...")
 - Tlaltecuhtli (/client): Click(the fluid duct (/obj/machinery/duct), the floor (82,128,2) (/turf/open/floor/wood), "mapwindow.map", "icon-x=21;icon-y=16;left=1;scr...")
 - 
```

@Time-Green

**Comment by Time-Green** [comment]
Yikes. I’ll get on it

**Comment by Time-Green** [comment]
Alright, false alarm, kinda.

Basically when ducts enter ridicilous lengths, BYOND mistakes it for infinite recursion and cuts it off. I managed to reproduce it on local and all it does is fuck up the ducts since alot of it is skipped.

I actually turned off loop_checks and everything went perfectly fine, with no spikes in performances or anything for that matter. Ducts even behaved like they should since it didn't get stopped. 

I do think I know why it scares byond and it's probably fixed by simply running them asynchronously so it's not one ducts single iteration of a for loop taking relatively long. 

I have something else I need to take care of, it'll probably be about tonight when I'll start making attempts to fix it.
